new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItMiiAddCluster.java

```diff
@@ -0,0 +1,998 @@
+    assertTrue(repilcaPatched, "patchDomainCustomResource(repilcas) failed");
+
+    patchStr = new StringBuffer("[{");
+    patchStr.append(" \"op\": \"replace\",")
+        .append(" \"path\": \"/spec/restartVersion\",")
+        .append(" \"value\": \"2\"")
+        .append(" }]");
+    logger.log(Level.INFO, "Restart version patch string: {0}", patchStr);
+
+    patch = new V1Patch(new String(patchStr));
+    boolean rvPatched = assertDoesNotThrow(() ->
+            patchDomainCustomResource(domainUid, domainNamespace, patch, "application/json-patch+json"),
+        "patchDomainCustomResource(restartVersion)  failed ");
+    assertTrue(rvPatched, "patchDomainCustomResource(restartVersion) failed");
+    logger.info("Snooze for 2 minutes for introspector to kick off");
+    try {
+      TimeUnit.MINUTES.sleep(2);
+    } catch (java.lang.InterruptedException ie) {
+      logger.info("Got InterruptedException during Thread.sleep");
+      fail("Got InterruptedException during Thread.sleep");
+    }
+    // Verify a rolling restart is triggered in a sequential fashion 
+    // admin-server --> managed-server1 --> managed-server2 
+    logger.info("Check admin server pod {0} to be restarted in ns {1}",
+        adminServerPodName, domainNamespace);
+    checkPodCreated(adminServerPodName, domainUid, domainNamespace);
+    checkPodReady(adminServerPodName, domainUid, domainNamespace);
+    checkServiceCreated(adminServerPodName, domainNamespace);
+    checkServerReadyStatusByExec(adminServerPodName, domainNamespace);
+
+    // check managed server pods restarted sequentially
+    for (int i = 1; i <= replicaCount; i++) {
+      logger.info("Wait for managed server pod {0} to be restarted in ns {1}",
+          managedServerPrefix + i, domainNamespace);
+      checkPodCreated(managedServerPrefix + i, domainUid, domainNamespace);
+      checkPodReady(managedServerPrefix + i, domainUid, domainNamespace);
+      checkServiceCreated(managedServerPrefix + i, domainNamespace);
+    }
+    // The ServerNamePrefix for the new dynamic cluster is dynamic-server
+    // Make sure the managed server from the new cluster is running
+    //
+    String newServerPodName = domainUid + "-dynamic-server1";
+    checkPodCreated(newServerPodName, domainUid, domainNamespace);
+    checkPodReady(newServerPodName, domainUid, domainNamespace);
+    checkServiceCreated(newServerPodName, domainNamespace);
+
+    boolean isPodRestarted  =
+        assertDoesNotThrow(() -> isPodRestarted(adminServerPodName, domainUid, domainNamespace, adminPodCreationTime),
+            String.format("Check if the admin pod is restarted"));
+    assertTrue(isPodRestarted, "Admin pod has not been restarted as expected");
+
+    oracle.weblogic.kubernetes.utils.ExecResult result = null;
+    int adminServiceNodePort = getAdminServiceNodePort(adminServerPodName + "-external", null, domainNamespace);
+    try {
+      checkCluster = new StringBuffer("status=$(curl --user weblogic:welcome1 ");
```

Comment:
this code is duplicated in all test methods, can be refactored..next PR is fine for this change

Reply:
Skipping to next round, when I will  use new Assertion to check the rolling restart